### ISSUE TITLE
W1b: an identity transform is not the identity at the class default `align_corners=False`

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4411]
Extends #3904 to the augmentation surface. Three geometric augmentation classes default to `align_corners=False` and delegate to a grid-normalising warp, and at that default a transform that is mathematically the identity does not return the input. `RandomPerspective(distortion_scale=0.)` draws `start_points == end_points`, `RandomElasticTransform(alpha=(0., 0.))` applies a zero displacement and `RandomThinPlateSpline(scale=1e-06)` a near-zero one — and all three move the image. Setting `align_corners=True` on the same call makes all three exact.

The defect is not in the classes. `warp_perspective` with `torch.eye(3)` shows the same behavior on the same fixture, while `warp_affine` with an identity `(2, 3)` matrix is exact under both flag values: the homography is normalised with the corner-aligned formula regardless of the flag, which is what #3904 reports and what PR #3945 proposes to fix. The classes inherit it because `warp_perspective`, `warp_image_tps` and `elastic_transform2d` all normalise coordinates the same way, and because their own default is the affected one.

The magnitude is fixture-bound — the underlying error is a half-pixel offset of the sampling grid, so it scales with the input's value range rather than being a fixed epsilon. On the row-major ramp below, whose values run from zero to just under one, the displacement moves a pixel by most of that range. That is large enough that a "disabled" augmentation at `p=1.0` and zero magnitude is not a no-op, which is how the affected classes behave in an `AugmentationSequential` whose parameters have been annealed to zero.

## Repro

```python
import torch, kornia, kornia.augmentation as K
from kornia.geometry.transform import warp_perspective, warp_affine
print("kornia", kornia.__version__, "torch", torch.__version__)
ramp = torch.arange(48., dtype=torch.float32).reshape(1, 1, 6, 8) / 48.   # 6 x 8, asymmetric

for name, make in (
    ("RandomPerspective(distortion_scale=0.)", lambda ac: K.RandomPerspective(distortion_scale=0., p=1., align_corners=ac)),
    ("RandomElasticTransform(alpha=(0., 0.))", lambda ac: K.RandomElasticTransform(alpha=(0., 0.), p=1., align_corners=ac)),
    ("RandomThinPlateSpline(scale=1e-06)   ", lambda ac: K.RandomThinPlateSpline(scale=1e-06, p=1., align_corners=ac)),
):
    out = {}
    for ac in (True, False):
        torch.manual_seed(0)
        out[ac] = round((make(ac)(ramp) - ramp).abs().max().item(), 6)
    print(f"{name}  max|out - in|:  align_corners=True {out[True]}   align_corners=False {out[False]}"
          f"   <- class default is {make(False).flags['align_corners']}")

torch.manual_seed(0)
p0 = K.RandomPerspective(distortion_scale=0., p=1.)
p0(ramp)
print("start_points == end_points for distortion_scale=0.:",
      torch.equal(p0._params["start_points"], p0._params["end_points"]))

# one level down: the same split between the two warps
H = torch.eye(3)[None]
M = torch.eye(3)[None, :2]
for ac in (True, False):
    print(f"warp_perspective(identity H, align_corners={ac}) max|out - in| ="
          f" {round((warp_perspective(ramp, H, (6, 8), align_corners=ac) - ramp).abs().max().item(), 6)}"
          f"   warp_affine(identity M) = {round((warp_affine(ramp, M, (6, 8), align_corners=ac) - ramp).abs().max().item(), 6)}")
```

## Observed

```
kornia 0.9.0rc1 torch 2.14.0
RandomPerspective(distortion_scale=0.)  max|out - in|:  align_corners=True 0.0   align_corners=False 0.734375   <- class default is False
RandomElasticTransform(alpha=(0., 0.))  max|out - in|:  align_corners=True 0.0   align_corners=False 0.734375   <- class default is False
RandomThinPlateSpline(scale=1e-06)     max|out - in|:  align_corners=True 1e-06   align_corners=False 0.734374   <- class default is False
start_points == end_points for distortion_scale=0.: True
warp_perspective(identity H, align_corners=True) max|out - in| = 0.0   warp_affine(identity M) = 0.0
warp_perspective(identity H, align_corners=False) max|out - in| = 0.734375   warp_affine(identity M) = 0.0
```

## Expected

Identity in, identity out, at both flag values — the coordinate normalisation should use the flag it is given, which is what PR #3945 implements for #3904. No augmentation-level change is needed once the warp is fixed; the three classes inherit the correction.

Until then the classes' own defaults are the reason users meet the defect without ever calling `warp_perspective`: `RandomPerspective`, `RandomElasticTransform` and `RandomThinPlateSpline` all default to the affected value while `warp_perspective`'s own default is `True`. The wider question of which default each augmentation should carry is separate and is tracked in the `align_corners` defaults issue for this batch.

## Classification

**Broken output** — a zero-magnitude transform returns a shifted image, on three public augmentation classes at their shipped defaults. Idiom: 3 (a guard whose predicate does not match the documented contract — the flag is accepted and then not consulted in the normalisation step).

## Disposition

`MERGE IN WINDOW — behavior change`. The fix lands in `kornia.geometry.transform`, not in `kornia.augmentation`, and it changes the pixels every current `align_corners=False` warp produces — it belongs with #3904 and PR #3945. Batch 6 records the executed defaults and the identity-warp behavior on the three classes, pins them, and points at #3904; it does not repair.

## Environment

kornia 0.9.0rc1 @ `d6a2c138` (main), torch 2.14.0, python 3.11, cpu. Found during the augmentation conventions audit (batch 6, tracking issue #4407); the batch documents the behavior above as it is and pins it, and does not change it.

Posted on behalf of @ducha-aiki by Claude (Fable 5.1).
